### ISSUE TITLE
WCAG 2.0 Support

### ISSUE THREAD

**Issue by unknowndomain** [BookStackApp/BookStack#1320]
**Describe the feature you'd like**
Within the EU, Australia, Canada, Hong Kong, India, Israel, Japan, New Zeland, and the USA any government website, and in many cases publicly funded organisations like Universities or even all businesses are required under law to support WCAG 2.0 or a similar identical law.

Bookstack isn't far off WCAG 2.0 support, and obviously the content people put up must actually bother to make use of accesibility features, but there are a few things which Bookstack fails on, some within automated testing are related to contrast which can easily be fixed using a CSS override in the header.

**Describe the benefits this feature would bring to BookStack users**
The benefit would be improved accessibility for BookStack users with visual impairments, and that BookStack could be used in the public sector for government and publicly funded websites.

**Additional context**
One example of a simple change is that the search box at the top of the website doesn't have a label associated, this could be easily added into the markup.

Document language identification is another easy fix.

It doesn't seem that it would be hard to officially support WCAG 2.0 and produce some documentation on how to create content in BookStack that is WCAG compliant.

**Comment by ssddanbrown** [comment]
Thanks for the suggestion @unknowndomain. I agree that supporting accessibility requirements is a good idea for BookStack. 

No point actioning anything now due to the design changes, which are changing most of the views, but I have instead assigned this to the next release after the redesign. We'll just aim for baseline support to close this off then any additional or new accessibility concerns can be raised as a new issue.

Something to note though, While I'm happy to aim to achieve accessibility I cannot really support or guarantee full compliance, for such public or government uses, due to the nature of the project being done in free time & non-funded.

**Comment by ssddanbrown** [comment]
Well this has been an interesting thing to work on. A lot of learnings that'll likely change how I do a lot of UI work going forward. Work done for this in the following commits:

* 5979f6667b7810bd2a574b83c54512e04e5b01cd
* 64abe10dc4a19128c89864367870bfb249077758
* 2dfe6c2d56719e03f31c74cc1d6d65e2df686031
* 9fbef8cd1be72cf5ee6589c38013ce615a919f7b
* cf5d51e7b8c9172906f7943d2d70b6b37801a2c4
* b27a5c7fb876c74c676eb9114383c25d82eb87fb
* 1b33a0c5b92a6e3cf8d75ce18edbac5311085a91

There have been quite a few design, functional and code changes to enhance accessibility. I discovered many dynamic elements in BookStack were not keyboard navigable, Most of which have been updated. The core layout elements have changed to better use HTML that indicates purpose. The default BookStack colors are also changing to meet Level A standards. 

Things are not fully compliant, but we're a lot further along now. I have updated the project readme with information on accessibility. The main troublesome spot I think is the page editors themselves but I'll do a deeper review when we look to better align the two editors. As said above, As it stands I won't really be able to guarantee a level of compliance but we can definitely keep it in mind, make targeted improvements and review parts of the system every so often. 

I'll close this issue off now since large general overall improvements have been made. For further improvements I'll allow focused issues to be raised.

These changes will be part of the next release, V0.27.
